oauth2_http/java/com/google/auth/oauth2/UserAuthorizer.java

```diff
@@ -181,6 +185,10 @@ public URL getAuthorizationUrl(String userId, String state, URI baseUri) {
       url.put("login_hint", userId);
     }
     url.put("include_granted_scopes", true);
+    if (pkce != null) {
```

Comment:
Do we need to check both `pkce.getCodeChallenge()` and `pkce.getCodeChallengeMethod()` not null for the case of user passing in their own PKCEProvider

Reply:
If those methods return NULL should the program continue to operate or throw an exception? 

Seems erroneous to have provided a PKCEProvider that is returning NULL.

Reply:
If a user provides custom PKCEProvider, I think we should validate it and throw an exception. Because we know if they pass in a PKCEProvider, they intend to use PKCE in fetch token call. Silently using plain or skipping PKCE in case of erroneous PKCEProvider will give them false impression of successful PKCE call. On another thought, do we need to give users the capability to pass in their own PKCEProvider? Maybe we can follow the Python implementation to always use DefaultPKCEProvider.

Reply:
I think it is still useful in case someone needs to disable it.

I've updated the code to include a null check in the builder